StortSpelProjekt/Game/src/Gamefiles/Components/HealthComponent.cpp

```diff
@@ -93,7 +93,7 @@ void component::HealthComponent::TakeDamage(int damage)
 		m_pParent->GetComponent<component::UpgradeComponent>()->OnDamage();
 	}
 
-	ChangeHealth((damage - m_FlatDamageReduction) * m_ProcentileDamageReduction); //Flat Damage gets applied first followed by multaplicative damage
+	ChangeHealth((-damage - m_FlatDamageReduction) * m_ProcentileDamageReduction); //Flat Damage gets applied first followed by multaplicative damage
```

Comment:
Flat damage reduction is damage removed flat, if it is removed before % reduction, you do not get 10 damage reduction if you have 10 flat damage reduction.

Reply:
Except you do if damage is 20, damge reduction is 10 and you have 50% you get.

(20-10) * 0.5 = 10 * 0.5 = 5.

you get a total 15 damage reduction. If you dont have the multiplicative you still get 10. It is done in this order to help prevent negative values. Because if you apply multaplicative first you can super easily go negative. also its to powerful